crates/common_utils/src/pii.rs

```diff
@@ -88,20 +98,103 @@ where
 }
 
 /// Email address
-#[derive(Debug)]
-pub struct Email;
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq, Eq, Default)]
+pub struct Email(
+    Secret<String>
+);
+
+impl From<String> for Email {
+    fn from(value: String) -> Self {
+        Email(Secret::new(value))
+    }
+}
 
-impl<T> Strategy<T> for Email
+impl<T> AppearsOnTable<T> for Email
+{
+}
+
+impl QueryFragment<Pg> for Email {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        // out.push_sql(" sql statement ");
+        Ok(())
+    }
+}
+
+
+impl Expression for Email {
+    type SqlType = sql_types::Nullable<Text>;
+}
```

Comment:
Do you need any of these implementations? As for `Expression`, you should be able to implement `AsExpression` instead, similar to how it's been done for `Secret` in the `masking` crate.

Reply:
```
error[E0277]: the trait bound `Email: AppearsOnTable<schema::customers::table>` is not satisfied
  --> crates/storage_models/src/customers.rs:48:33
   |
48 | #[derive(Clone, Debug, Default, AsChangeset, router_derive::DebugAsDisplay)]
   |                                 ^^^^^^^^^^^ the trait `AppearsOnTable<schema::customers::table>` is not implemented for `Email`
```
AppearsOnTable is required for Email

Reply:
You can derive `AsExpression` on `Email` like so:

```rust
#[derive(diesel::AsExpression)]
#[diesel(sql_type = diesel::sql_types::Text)]
pub struct Email(Secret<String>);
```

This should handle it, since `AsExpression` helps with obtaining an `Expression`, and `AppearsOnTable` is implemented automatically for types which implement `Expression`. I did something similar for `HashedApiKey` when working on #511. (You can check [this snippet](https://github.com/juspay/hyperswitch/blob/b12762e776205a1b6375a7c49447e8a3dc541b8a/crates/storage_models/src/api_keys.rs#L80-L82) and the [`diesel_impl` module](https://github.com/juspay/hyperswitch/blob/b12762e776205a1b6375a7c49447e8a3dc541b8a/crates/storage_models/src/api_keys.rs#L96).)